### PR TITLE
Gem jar divya

### PR DESCRIPTION
**Pull request by gem-divyamadan**
comments removed as suggested.

### CODE REVIEW

File: src/main/java/com/qa/mis/locators/DashboardProfileLocator.java [hunk 1]
```diff
@@ -49,8 +50,9 @@ public class DashboardProfileLocator {
 
     public static By fromDate=By.xpath("(//input[@name='fromDate'])[1]");
 
-    public static By selectFromDate=By.xpath("(//td[@class='day'])[1]");
-
+  public static By selectNextArrow=By.xpath("//body[1]/div[16]/div[1]/table[1]/thead[1]/tr[2]/th[3]");
```
Comment (Gem-NeerajSingh) [inline src/main/java/com/qa/mis/locators/DashboardProfileLocator.java:53]: use dynamic xpath  , Avoid using absolute xpaths

File: src/main/java/com/qa/mis/stepdefinition/DashboardProfileSteps.java [hunk 2]
```diff
@@ -321,18 +386,93 @@ public void clickOnPincode() {
...
+        } catch (Exception e) {
+            GemTestReporter.addTestStep("ERROR", "password verification error" + e, STATUS.FAIL);
+        }
+    }
+
+    @When("click on edit details button")
+    public void clickOnEditDetailsButton() {
+
+        DriverAction.waitSec(5);
+        DriverAction.click(clickOnChangeDetails, "click on edit details button");
```
Comment (Gem-NeerajSingh) [inline src/main/java/com/qa/mis/stepdefinition/DashboardProfileSteps.java:411]: use element visibility check or  try catch

File: src/main/java/com/qa/mis/stepdefinition/DashboardProfileSteps.java [hunk 3]
```diff
@@ -321,18 +386,93 @@ public void clickOnPincode() {
...
+    }
+
+    @When("click on change details button")
+    public void clickOnChangeDetailsButton() {
+
+        DriverAction.waitSec(5);
+        DriverAction.click(clickOnChangeDetails, "click on edit details button");
```
Comment (Gem-NeerajSingh) [inline src/main/java/com/qa/mis/stepdefinition/DashboardProfileSteps.java:418]: use element visibility check or  try catch

File: src/main/java/com/qa/mis/stepdefinition/DashboardProfileSteps.java [hunk 4]
```diff
@@ -321,18 +386,93 @@ public void clickOnPincode() {
...
+
+        DriverAction.waitSec(5);
+        DriverAction.typeText(enterMobileNo, "123456789");
+    }
+
+    @And("click on update mobile number")
+    public void clickOnUpdateMobileNumber() {
+
+        DriverAction.waitSec(5);
+        DriverAction.click(updateMobileNo, "click on update mobile no. button");
```
Comment (Gem-NeerajSingh) [inline src/main/java/com/qa/mis/stepdefinition/DashboardProfileSteps.java:432]: use element visibility check or  try catch

File: src/main/java/com/qa/mis/stepdefinition/DashboardProfileSteps.java [hunk 5]
```diff
@@ -321,18 +386,93 @@ public void clickOnPincode() {
...
+        } catch (Exception e) {
+            GemTestReporter.addTestStep("ERROR", "invalid ph no verification" + e, STATUS.FAIL);
+        }
+    }
+
+    @And("enter extension number")
+    public void enterExtensionNumber() {
+
+        DriverAction.waitSec(5);
+        DriverAction.typeText(enterExtNo, "11");
```
Comment (Gem-NeerajSingh) [inline src/main/java/com/qa/mis/stepdefinition/DashboardProfileSteps.java:455]: use element visibility check or  try catch

File: src/main/java/com/qa/mis/stepdefinition/ReferalStep.java [hunk 6]
```diff
@@ -31,83 +38,108 @@ public void enterPassword() {
...
-        DriverAction.waitUntilElementAppear(signIn, 2);
-        DriverAction.click(signIn);
-        DriverAction.waitSec(10);
+        try {
+            //DriverAction.waitUntilElementAppear(signIn, 2);
+            DriverAction.waitSec(5);
+            DriverAction.click(signIn);
+            DriverAction.waitSec(10);
+        } catch (Exception e) {
+            GemTestReporter.addTestStep("ERROR", "SignIn error" + e, STATUS.FAIL, DriverAction.takeSnapShot());
```
Comment (Gem-NeerajSingh) [inline src/main/java/com/qa/mis/stepdefinition/ReferalStep.java:47]: please use proper error logging

File: src/main/java/com/qa/mis/stepdefinition/ReferalStep.java [hunk 7]
```diff
@@ -116,51 +148,171 @@ public void referralWindowIsVisible() {
...
+
     }
 
     @And("^click on email under referral$")
     public void clickOnEmailUnderReferral() {
-        DriverAction.waitUntilElementAppear(refEmail,3);
-        DriverAction.click(refEmail,"referral email text");
+        //DriverAction.waitUntilElementAppear(refEmail, 3);
+        DriverAction.waitSec(5);
+        DriverAction.click(refEmail, "referral email text");
```
Comment (Gem-NeerajSingh) [inline src/main/java/com/qa/mis/stepdefinition/ReferalStep.java:176]: use element visibility check or  try catch

File: src/main/java/com/qa/mis/stepdefinition/ReferalStep.java [hunk 8]
```diff
@@ -116,51 +148,171 @@ public void referralWindowIsVisible() {
...
     }
 
     @And("^enter the email address$")
     public void enterTheEmailAddress() {
-        DriverAction.waitUntilElementAppear(refEmail,2);
-        DriverAction.typeText(refEmail,"abcde");
+
+        DriverAction.waitSec(5);
+        DriverAction.typeText(refEmail, "abcde");
```
Comment (Gem-NeerajSingh) [inline src/main/java/com/qa/mis/stepdefinition/ReferalStep.java:183]: use element visibility check or  try catch

File: src/main/java/com/qa/mis/stepdefinition/ReferalStep.java [hunk 9]
```diff
@@ -116,51 +148,171 @@ public void referralWindowIsVisible() {
...
     }
 
     @And("^click on save$")
     public void clickOnSave() {
-        DriverAction.waitUntilElementAppear(refSaveButton,2);
-        DriverAction.click(refSaveButton,"click on save button under referral");
+
+        DriverAction.waitSec(5);
+        DriverAction.click(refSaveButton, "click on save button under referral");
```
Comment (Gem-NeerajSingh) [inline src/main/java/com/qa/mis/stepdefinition/ReferalStep.java:190]: use element visibility check or  try catch

**Review by Gem-NeerajSingh — CHANGES_REQUESTED** [review]
Changes required

File: src/main/java/com/qa/mis/locators/DashboardProfileLocator.java [hunk 10]
```diff
@@ -49,8 +49,12 @@ public class DashboardProfileLocator {
     public static By passwordNotMatch=By.xpath("//span[@id='password_match']");
 
     public static By fromDate=By.xpath("(//input[@name='fromDate'])[1]");
+    // ("body.with-side-menu.control-panel.control-panel-compact.chrome-browser.modal-open:nth-child(2) div.modal.fade.in:nth-child(10) div.modal-dialog.modal-lg div.modal-content:nth-child(1) div.modal-body.lunch div.row:nth-child(1) div.col-md-4:nth-child(1) div.form-group div.inputGroupContainer.date.date-picker div.input-group span.input-group-btn button.btn.default > i.fa.fa-calendar");
 
-  public static By selectNextArrow=By.xpath("//body[1]/div[16]/div[1]/table[1]/thead[1]/tr[2]/th[3]");
+
+
+
+    public static By selectNextArrow=By.xpath("//body[1]/div[16]/div[1]/table[1]/thead[1]/tr[2]/th[3]");
```
Comment (Gem-NeerajSingh) [inline src/main/java/com/qa/mis/locators/DashboardProfileLocator.java:57]: Use dynamic xpath (Avoid using absolute xpath)

**Review by Gem-NeerajSingh — APPROVED** [review]

File: src/main/java/com/qa/mis/stepdefinition/DashboardProfileSteps.java [hunk 11]
```diff
@@ -38,54 +39,59 @@ public void enterPassword() {
...
     public void signIn() {
-        // DriverAction.waitUntilElementAppear(signIn, 2);
+       try{
         DriverAction.waitSec(5);
         DriverAction.click(signIn);
-        //DriverAction.waitSec(10);
-    }
+
+    }catch (Exception e){
+           GemTestReporter.addTestStep("ERROR", "not found SignIn" + e, STATUS.FAIL,DriverAction.takeSnapShot());
```
Comment (Gem-NeerajSingh) [inline src/main/java/com/qa/mis/stepdefinition/DashboardProfileSteps.java:47]: Please leave proper error logging

File: src/main/java/com/qa/mis/stepdefinition/DashboardProfileSteps.java [hunk 12]
```diff
@@ -38,54 +39,59 @@ public void enterPassword() {
...
+       }
```
Comment (Gem-NeerajSingh) [inline src/main/java/com/qa/mis/stepdefinition/DashboardProfileSteps.java:48]: please use proper error logging

File: src/main/java/com/qa/mis/stepdefinition/ReferalStep.java [hunk 13]
```diff
@@ -167,28 +168,43 @@ public void userChecksReferralSection() {
 
             GemTestReporter.addTestStep("verifyJdWindow", "verify window is visible", STATUS.FAIL, DriverAction.takeSnapShot());
 
+    }catch(Exception e){
+            GemTestReporter.addTestStep("ERROR", "not found verify JD window" + e, STATUS.FAIL, DriverAction.takeSnapShot());
```
Comment (Gem-NeerajSingh) [inline src/main/java/com/qa/mis/stepdefinition/ReferalStep.java:172]: avoid shortforms in error logging

**Review by Gem-NeerajSingh — CHANGES_REQUESTED** [review]
PFB

**Comment by gem-divyamadan** [comment]
Done

**Pull request closed without merging**